extensions/agroal/runtime/src/main/java/io/quarkus/agroal/runtime/vault/VaultDataSourceRuntimeConfig.java

```diff
@@ -0,0 +1,30 @@
+package io.quarkus.agroal.runtime.vault;
+
+import static io.quarkus.agroal.VaultDbService.PASSWORD_PROPERTY_NAME;
+
+import java.util.Optional;
+
+import io.quarkus.runtime.annotations.ConfigGroup;
+import io.quarkus.runtime.annotations.ConfigItem;
+
+@ConfigGroup
+public class VaultDataSourceRuntimeConfig {
```

Comment:
You have `quarkus.datasource.vault.path` and `quarkus.datasource.dynamic.vault.dbrole` documented above, what are the fully qualified names of the below properties ? It feels that this config group should reside in the `vault` extension ?

Reply:
well these are properties of the datasource.

so when you configure your ds you can do it the usual way:
```
quarkus.datasource.url=jdbc:h2:tcp://localhost/mem:test_quarkus;DB_CLOSE_DELAY=-1
quarkus.datasource.username=sa
quarkus.datasource.password=sa
```

or fetch the password from the kv store at path databases/myapp_db with key='password':
```
quarkus.datasource.url=jdbc:postgresql://localhost:5432/mydb
quarkus.datasource.username=postgres
quarkus.datasource.vault.path=databases/myapp_db
```

or get the user and the password from vault using dynamic credentials:
```
quarkus.datasource.url=jdbc:postgresql://localhost:5432/mydb
quarkus.datasource.vault.dbrole=mydbrole
```

if we wanted to hide that, that would imply more indirection. for instance we could have in the config some credential providers:
```
# credential provider 'foo'
# static credentials returning key='password' at vault path 'databases/myapp_db'
quarkus.vault.credential-provider.foo.type=kv
quarkus.vault.credential-provider.foo.path=databases/myapp_db
quarkus.vault.credential-provider.foo.key=password

# credential provider 'bar'
# returning dynamically generated credentials for db role='mydbrole'
```
quarkus.vault.credential-provider.bar.type=db
quarkus.vault.credential-provider.bar.dbrole=mydbrole
```

then in the datasource configuration you could do:
```
# using static credentials
quarkus.datasource.url=jdbc:postgresql://localhost:5432/mydb
quarkus.datasource.username=postgres
quarkus.datasource.credential-provider=foo
```
or
```
# using dynamic credentials
quarkus.datasource.url=jdbc:postgresql://localhost:5432/mydb
quarkus.datasource.credential-provider=bar
```

and in the spi we would have just one interface:
```
public interface CredentialProvider {
  Properties getCredentials(String credentialProviderName);
}
```

does it make sense?